stablehlo/reference/Ops.cpp

```diff
@@ -157,6 +157,25 @@ Tensor evalOrOp(const Tensor &lhs, const Tensor &rhs, Type resultType) {
   return result;
 }
 
+Tensor evalPadOp(const Tensor &operand, const Tensor &paddingValue,
+                 ArrayRef<int64_t> edgePaddingLow,
+                 ArrayRef<int64_t> interiorPadding, Type resultType) {
+  Tensor result(resultType);
+  for (auto resultItr = result.index_begin(); resultItr != result.index_end();
+       ++resultItr)
+    result.set(*resultItr, paddingValue.get({}));
+  for (auto operandItr = operand.index_begin();
+       operandItr != operand.index_end(); ++operandItr) {
+    SmallVector<int64_t> resultIdx(result.getType().getRank());
+    for (auto i = 0; i < operand.getType().getRank(); ++i)
+      resultIdx[i] =
+          edgePaddingLow[i] + (*operandItr)[i] * (interiorPadding[i] + 1);
```

Comment:
Something to consider in the future - I think it might be useful to have `IndexSpaceIterator` return `Index` instead of `ArrayRef<int64_t>`, and then have this `Index` class overload operators like `*` or `+`. cc @sdasgup3.